include/invert_laplace.hxx

```diff
@@ -192,7 +192,7 @@ public:
    * 
    * @param[in] opt  The options section to use. By default "laplace" will be used
    */
-  static Laplacian *create(Options *opt = nullptr, const CELL_LOC loc = CELL_CENTRE, Mesh *mesh_in = mesh);
+  static Laplacian *create(Options *opt = nullptr, const CELL_LOC loc = CELL_CENTRE, Mesh *mesh_in = bout::globals::mesh);
```

Comment:
Should `mesh_in` really be `nullptr` here? Presumably this is passed to the constructor noted above (or equivalent) which can then choose what to do if it receives `nullptr`?

Reply:
Agreed -- `nullptr` is a better default, it's what we do in other bits of the code, as well as giving us some space to potential remove the global mesh entirely.

Reply:
We can't just pass `nullptr` here because `LaplaceFactory` uses `mesh_in` to check whether we can use a serial solver
https://github.com/boutproject/BOUT-dev/blob/08cc9f225a207c96bd54aacaa6355ac7f7b4d98e/src/invert/laplace/laplacefactory.cxx#L46-L47
I think this was originally to allow the default Laplace solver to switch between `serial_tri` and `spt` depending on whether a serial or parallel solver was needed, but now the default is `cyclic` in either case. I think it would be better now to allow any Laplace solver to be created, and move the check into the constructor of the serial solvers. Does that sound reasonable, and should I do it here or create a separate PR?

Reply:
If `LaplaceFactory` gets a `nullptr`, it could just fetch the global mesh though?

`LaplaceFactory` should get rewritten anyway (e.g. using `GenericFactory`), so I would leave that to another PR